### PR TITLE
[Techinal Debt][NO TESTS NEEDED] Cleanup ABI vs. Tunnel CLI commands

### PR DESCRIPTION
**Pull request by jwhonce**
This commit cleans up two issues:
* Most commands support all EngineModes so default to that. Let
  outlayers declare their intent.
* Use cobra.Annotations to set supported EngineMode. This simplies
  instantiating commands as there is now one method to communicate a
  commands requirements rather than two.

Additionally:
* Combined aliased commands into one file
* Fixed aliased commands where Args field did not match
* Updated examples in README.md for writing commands

Signed-off-by: Jhon Honce <jhonce@redhat.com>

### CODE REVIEW

File: cmd/podman/containers/mount.go [hunk 1]
```diff
@@ -25,27 +25,28 @@ var (
 `
 
 	mountCommand = &cobra.Command{
+		Annotations: map[string]string{
+			registry.UnshareNSRequired: "",
+			registry.ParentNSRequired:  "",
+			registry.EngineMode:        registry.ABIMode,
```
Comment (mheon) [inline cmd/podman/containers/mount.go:31]: You sure about this one? I remember losing an argument with @baude that this one should be ABI-only

Comment (rhatdan) [reply cmd/podman/containers/mount.go:31]: Yes mount makes no sense on a remote server.

Comment (jwhonce) [reply cmd/podman/containers/mount.go:31]: @mheon The command was limited to just ABI support and continues to be so. Not listing an EngineMode means the command supports all modes.

File: cmd/podman/main.go [hunk 2]
```diff
@@ -42,38 +43,41 @@ func main() {
...
-					}
-				} else {
-					_, found = c.Command.Annotations[registry.ParentNSRequired]
-					if rootless.IsRootless() && found {
-						c.Command.RunE = func(cmd *cobra.Command, args []string) error {
-							return fmt.Errorf("cannot run command %q in rootless mode", cmd.CommandPath())
-						}
-					}
+		if supported, found := c.Command.Annotations[registry.EngineMode]; found {
+			if !strings.Contains(cfg.EngineMode.String(), supported) {
```
Comment (Luap99) [inline cmd/podman/main.go:47]: Why contains? Should this be `cfg.EngineMode.String() != supported`?

File: cmd/podman/images/list.go [hunk 3]
```diff
@@ -49,6 +48,19 @@ var (
...
+		Use:               "images [options] [IMAGE]",
+		Args:              imageListCmd.Args,
+		Short:             imageListCmd.Short,
+		Long:              imageListCmd.Long,
+		RunE:              imageListCmd.RunE,
+		ValidArgsFunction: imageListCmd.ValidArgsFunction,
+		Example: `podman images --format json
+  podman images --sort repository --format "table {{.ID}} {{.Repository}} {{.Tag}}"
+  podman images --filter dangling=true`,
+		DisableFlagsInUseLine: true,
```
Comment (Luap99) [inline cmd/podman/images/list.go:61]: DisableFlagsInUseLine can be removed it is set for all commands in main

File: cmd/podman/main.go (hunk 2 above)
Comment (jwhonce) [reply cmd/podman/main.go:47]: @Luap99 I did contains to support any future engines.

File: cmd/podman/containers/unmount.go [hunk 4]
```diff
@@ -20,11 +20,12 @@ var (
   An unmount can be forced with the --force flag.
 `
 	unmountCommand = &cobra.Command{
-		Use:     "unmount [options] CONTAINER [CONTAINER...]",
-		Aliases: []string{"umount"},
-		Short:   "Unmounts working container's root filesystem",
-		Long:    description,
-		RunE:    unmount,
+		Annotations: map[string]string{registry.EngineMode: registry.ABIMode},
```
Comment (rhatdan) [inline cmd/podman/containers/unmount.go:23]: Is this saying unmount is supported in podman-remote?

Comment (jwhonce) [reply cmd/podman/containers/unmount.go:23]: No, only "local"

Comment (rhatdan) [reply cmd/podman/containers/unmount.go:23]: What I find confusing is that Annotations indicates which engines to use, now rather then Mode, is that basically the change that is going on?

File: cmd/podman/images/inspect.go [hunk 5]
```diff
@@ -12,6 +12,7 @@ import (
 var (
 	// Command: podman image _inspect_
 	inspectCmd = &cobra.Command{
+
```
[the remark this answers is not in the document]
Comment (jwhonce) [reply cmd/podman/images/inspect.go:15]: Global sed bit me...

**Comment by TomSweeneyRedHat** [comment]
LGTM
and that's some serious sandbox scrubbing @jwhonce !
All happy green test buttons and a shiny merge button with another LGTM

**Pull request merged**